monai/transforms/utils.py

```diff
@@ -149,22 +149,26 @@ def zero_margins(img: np.ndarray, margin: int) -> bool:
 
 
 def rescale_array(
-    arr: NdarrayOrTensor, minv: float = 0.0, maxv: float = 1.0, dtype: Union[DtypeLike, torch.dtype] = np.float32
+    arr: NdarrayOrTensor,
+    minv: float = 0.0,
+    maxv: float = 1.0,
+    dtype: Optional[Union[DtypeLike, torch.dtype]] = np.float32,
 ) -> NdarrayOrTensor:
     """
     Rescale the values of numpy array `arr` to be from `minv` to `maxv`.
     """
-    if dtype is not None:
-        arr, *_ = convert_data_type(arr, dtype=dtype)
-
+    dtype_ = dtype or arr.dtype
     mina = arr.min()
     maxa = arr.max()
 
     if mina == maxa:
         return arr * minv
 
     norm = (arr - mina) / (maxa - mina)  # normalize the array first
-    return (norm * (maxv - minv)) + minv  # rescale by minv and maxv, which is the normalized array by default
```

Comment:
Fixed a bug here that we should convert data type after computation.

Reply:
this causes an issue in https://github.com/Project-MONAI/MONAI/pull/3242
```
======================================================================

ERROR: test_blend_2 (tests.test_blend_images.TestBlendImages)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/opt/conda/lib/python3.6/site-packages/parameterized/parameterized.py", line 533, in standalone_func

    return func(*(a + p.args), **p.kwargs)

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_blend_images.py", line 39, in test_blend

    blended = blend_images(image[None], label[None])

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/visualize/utils.py", line 144, in blend_images

    label = rescale_array(label)

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/utils.py", line 167, in rescale_array

    norm = (arr - mina) / (maxa - mina)  # normalize the array first

RuntimeError: Integer division of tensors using div or / is no longer supported, and in a future release div will perform true division as in Python 3. Use true_divide or floor_divide (// in Python) instead.



======================================================================

ERROR: test_blend_3 (tests.test_blend_images.TestBlendImages)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/opt/conda/lib/python3.6/site-packages/parameterized/parameterized.py", line 533, in standalone_func

    return func(*(a + p.args), **p.kwargs)

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_blend_images.py", line 39, in test_blend

    blended = blend_images(image[None], label[None])

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/visualize/utils.py", line 144, in blend_images

    label = rescale_array(label)

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/utils.py", line 167, in rescale_array

    norm = (arr - mina) / (maxa - mina)  # normalize the array first

RuntimeError: Integer division of tensors using div or / is no longer supported, and in a future release div will perform true division as in Python 3. Use true_divide or floor_divide (// in Python) instead.



======================================================================

ERROR: test_blend_4 (tests.test_blend_images.TestBlendImages)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/opt/conda/lib/python3.6/site-packages/parameterized/parameterized.py", line 533, in standalone_func

    return func(*(a + p.args), **p.kwargs)

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_blend_images.py", line 39, in test_blend

    blended = blend_images(image[None], label[None])

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/visualize/utils.py", line 144, in blend_images

    label = rescale_array(label)

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/utils.py", line 167, in rescale_array

    norm = (arr - mina) / (maxa - mina)  # normalize the array first

RuntimeError: Integer division of tensors using div or / is no longer supported, and in a future release div will perform true division as in Python 3. Use true_divide or floor_divide (// in Python) instead.



======================================================================

ERROR: test_blend_5 (tests.test_blend_images.TestBlendImages)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/opt/conda/lib/python3.6/site-packages/parameterized/parameterized.py", line 533, in standalone_func

    return func(*(a + p.args), **p.kwargs)

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_blend_images.py", line 39, in test_blend

    blended = blend_images(image[None], label[None])

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/visualize/utils.py", line 144, in blend_images

    label = rescale_array(label)

  File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/transforms/utils.py", line 167, in rescale_array

    norm = (arr - mina) / (maxa - mina)  # normalize the array first

RuntimeError: Integer division of tensors using div or / is no longer supported, and in a future release div will perform true division as in Python 3. Use true_divide or floor_divide (// in Python) instead.



----------------------------------------------------------------------

```